lib/src/front_end/ast_node_visitor.dart

```diff
@@ -130,7 +130,17 @@ class AstNodeVisitor extends ThrowingAstVisitor<void>
 
   @override
   void visitAssertStatement(AssertStatement node) {
-    throw UnimplementedError();
+    token(node.assertKeyword);
+
+    var arguments = [node.condition];
+    if (node.message case var message?) arguments.add(message);
+    createList(
+      arguments,
+      leftBracket: node.leftParenthesis,
+      rightBracket: node.rightParenthesis,
+    );
```

Comment:
If you really wanted to be clever, you could do:

```dart
    createList(
      [
        node.condition,
        if (node.message case var message?) message
      ],
      leftBracket: node.leftParenthesis,
      rightBracket: node.rightParenthesis,
    );
```

Reply:
Oh that's kinda wild. Done.